sweepai/core/vector_db.py

```diff
@@ -265,23 +265,23 @@ def get_deeplake_vs_from_repo(
             if len(file_list) > MAX_FILES:
                 score_factors.append((1, 2, 5)) # This is a low score
                 continue
-            try:
-                cache_key = f"{repo_name}-{file_path}-{CACHE_VERSION}"
-                if cache_inst and cache_success:
-                    cached_value = cache_inst.get(cache_key)
-                    if cached_value:
-                        score_factor = json.loads(cached_value)
-                        score_factors.append(score_factor)
-                        continue
-                commits = list(repo.get_commits(path=file_path, sha=branch_name))
-                score_factor = get_factors(contents, commits)
-                if cache_inst and cache_success:
-                    cache_inst.set(cache_key, json.dumps(score_factor), ex=60 * 60 * 2)
-                score_factors.append(score_factor)
-            except Exception as e:
-                logger.warning(f"Received warning during scoring {e}, skipping...")
-                score_factors.append((1, 2, 5))
-                continue
+            def calculate_score_factor(repo_name, file_path, contents, cache_inst, cache_success, repo, branch_name):
```

Comment:
move the function definition completely out of the loop. also make sure calculate_score_factor happens after the loop terminates